src/schemas/json/qgoda.json

```diff
@@ -0,0 +1,445 @@
+              "modules": ["TT2", "Strip"]
+            },
+            "xml": {
+              "modules": ["TT2", "Strip"]
+            }
+          },
+          "description": "The processor chains.",
+          "patternProperties": {
+            "[_a-zA-Z][a-zA-Z0-9]*": {
+              "additionalProperties": false,
+              "description": "Properties of one processor chain.",
+              "properties": {
+                "modules": {
+                  "description": "The module names.",
+                  "items": {
+                    "type": "string"
+                  },
+                  "type": "array"
+                },
+                "suffix": {
+                  "description": "An optional suffix if different from original filename.",
+                  "type": "string"
+                },
+                "wrapper": {
+                  "description": "An optional wrapper for a second run.",
+                  "type": "string"
+                }
+              },
+              "required": ["modules"],
+              "type": "object"
+            }
+          },
+          "type": "object"
+        },
+        "options": {
+          "additionalProperties": true,
+          "default": {
+            "HTMLFilter": {
+              "AnchorTarget": {},
+              "CleanUp": {},
+              "Generator": {},
+              "TOC": {
+                "content-tag": "qgoda-content",
+                "end": 6,
+                "start": 2,
+                "template": "components/toc.html",
+                "toc-tag": "qgoda-toc"
+              }
+            }
+          },
+          "description": "Additional options for the processor plug-ins",
+          "type": "object"
+        },
+        "triggers": {
+          "default": {
+            "htm": "html",
+            "html": "html",
+            "json": "raw",
+            "md": "markdown",
+            "mdown": "markdown",
+            "mdwn": "markdown",
+            "mkd": "markdown",
+            "mkdn": "markdown",
+            "xml": "raw"
+          },
+          "description": "Filename extenders that trigger a  particular chain if not specified in front matter or defaults.",
+          "patternProperties": {
+            ".+": {
+              "description": "The filename extender.",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        }
+      },
+      "required": ["chains", "options", "triggers"],
+      "type": "object"
+    },
+    "scm": {
+      "const": "git",
+      "description": "Source code management (SCM) that is in use. If present, only files that are under version control and those matching 'no-scm' are processed. Currently only git is supported.",
+      "type": "string"
+    },
+    "site": {
+      "description": "Site-specific variables.  You can also choose the namespace 'private' if you prefer."
+    },
+    "srcdir": {
+      "const": "/Users/guidoflohr/qgoda/qgoda-site",
```

Comment:
Doesn't this [imply](https://json-schema.org/understanding-json-schema/reference/generic.html) that the only valid value is a directory that requires the user `guidoflohr` to be registers and active on the system? There are a few other such cases in this schema.

If this was done because the user themselves should not set the vaue in the config value, maybe you can use the `readOnly` property described [here](https://json-schema.org/understanding-json-schema/reference/generic.html)?

Reply:
Good catch! Thanks! I have used `readOnly` for the constant and changed the other value to ".", see https://www.qgoda.net/schemas/qgoda.json.